File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink/20260831_074008/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
)

type Record struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	
	// Read and skip header
	header, err := reader.Read()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading header: %v\n", err)
		os.Exit(1)
	}
	
	// We'll use the header to determine column indices
	// For simplicity, assume the columns are in the expected order: Name, Age, Email, Score
	// But let's be robust and use the header to map columns
	
	colIndex := make(map[string]int)
	for i, h := range header {
		colIndex[h] = i
	}
	
	nameIdx, _ := colIndex["Name"]
	ageIdx, _ := colIndex["Age"]
	emailIdx, _ := colIndex["Email"]
	scoreIdx, _ := colIndex["Score"]
	
	var records []Record
	
	for {
		row, err := reader.Read()
		if err != nil {
			if err.Error() == "EOF" {
				break
			}
			fmt.Fprintf(os.Stderr, "Error reading row: %v\n", err)
			break
		}
		
		if len(row) <= max(nameIdx, max(ageIdx, max(emailIdx, scoreIdx))) {
			continue
		}
		
		age, err := strconv.Atoi(row[ageIdx])
		if err != nil {
			age = 0
		}
		
		score, err := strconv.ParseFloat(row[scoreIdx], 64)
		if err != nil {
			score = 0.0
		}
		
		record := Record{
			Name:  row[nameIdx],
			Age:   age,
			Email: row[emailIdx],
			Score: score,
		}
		
		records = append(records, record)
	}
	
	output, err := json.Marshal(records)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}
	
	fmt.Println(string(output))
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink/20260831_074008/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const inputFile = path.join('input', 'data.csv');
const content = fs.readFileSync(inputFile, 'utf-8');

const lines = content.split('\n').filter(line => line.trim() !== '');

if (lines.length < 2) {
  console.log('[]');
  process.exit(0);
}

const header = lines[0].split(',').map(h => h.trim());

const result: Array<{ Name: string; Age: number; Email: string; Score: number }> = [];

for (let i = 1; i < lines.length; i++) {
  const row = lines[i].split(',').map(v => v.trim());
  if (row.length < header.length) continue;

  const name = row[0];
  const age = parseInt(row[1], 10);
  const email = row[2];
  const score = parseFloat(row[3]);

  result.push({
    Name: name,
    Age: age,
    Email: email,
    Score: score
  });
}

console.log(JSON.stringify(result, null, 2));